frontend/src/scenes/data-warehouse/editor/QueryWindow.tsx

```diff
@@ -0,0 +1,175 @@
+    }
+    const logic = hogQLQueryEditorLogic(hogQLQueryEditorLogicProps)
+    const { queryInput, promptError } = useValues(logic)
+    const { setQueryInput, saveQuery, saveAsView } = useActions(logic)
+
+    const codeEditorKey = `hogQLQueryEditor/${router.values.location.pathname}`
+
+    const codeEditorLogicProps: CodeEditorLogicProps = {
+        key: codeEditorKey,
+        sourceQuery: query,
+        query: queryInput,
+        language: 'hogQL',
+        metadataFilters: query.filters,
+        monaco,
+        editor,
+        multitab: true,
+    }
+    const { activeModelUri, allModels, hasErrors, error, isValidView } = useValues(
+        codeEditorLogic(codeEditorLogicProps)
+    )
+
+    const { createModel, setModel, deleteModel, setModels, addModel, updateState } = useActions(
+        codeEditorLogic(codeEditorLogicProps)
+    )
+
+    const modelKey = `hogQLQueryEditor/${activeModelUri?.path}`
+
+    useEffect(() => {
+        if (monaco && activeModelUri) {
+            const _model = monaco.editor.getModel(activeModelUri)
+            const val = _model?.getValue()
+            setQueryInput(val ?? '')
+            saveQuery()
+        }
+    }, [activeModelUri])
+
+    const onAdd = useCallback(() => {
+        createModel()
+    }, [createModel])
+
+    return (
+        <div className="flex flex-1 flex-col h-full">
+            <QueryTabs
+                models={allModels}
+                onClick={setModel}
+                onClear={deleteModel}
+                onAdd={onAdd}
+                activeModelUri={activeModelUri}
+            />
+            <QueryPane
+                queryInput={queryInput}
+                promptError={promptError}
+                codeEditorProps={{
+                    onChange: (v) => {
+                        setQueryInput(v ?? '')
+                        updateState()
+                    },
+                    onMount: (editor, monaco) => {
+                        setMonacoAndEditor([monaco, editor])
+
+                        const allModelQueries = localStorage.getItem(editorModelsStateKey(codeEditorKey))
+                        const activeModelUri = localStorage.getItem(activemodelStateKey(codeEditorKey))
+
+                        if (allModelQueries) {
+                            // clear existing models
+                            monaco.editor.getModels().forEach((model) => {
+                                model.dispose()
+                            })
+
+                            const models = JSON.parse(allModelQueries || '[]')
+                            const newModels: Uri[] = []
+
+                            models.forEach((model: Record<string, any>) => {
+                                if (monaco) {
+                                    const uri = monaco.Uri.parse(model.path)
+                                    const newModel = monaco.editor.createModel(model.query, 'hogQL', uri)
+                                    editor?.setModel(newModel)
```

Comment:
So the last saved model will be the one selected in the UI? May be good to break this out of this `forEach` loop